### DISCUSSION TITLE
The GitHub Skyline CLI extension is now available!

### DISCUSSION THREAD

**Discussion by chrisreddington** [community/community#146467] [Announcements]
Hey everyone!

As we wrap up 2024, I’m excited to share some news with you! We’ve [released `gh-skyline`](https://github.com/github/gh-skyline), a GitHub CLI extension that allows you to render your contribution graph as an STL file for 3D printing!

<img width="1248" alt="skyline-social" src="https://github.com/user-attachments/assets/60b5f5ea-3a26-4c24-8232-1e6026edc84d">

If you used our former GitHub Skyline site, then this concept may be familiar to you! But we’ve sprinkled a few additional features into the CLI as well, including:

📅 Customizable year selection: Show off a single year or flex with multi-year masterpieces. See an example below!

![github-skyline-example](https://github.com/user-attachments/assets/50828aa6-8e27-4785-ab34-26fd896e77d4)

🎨 ASCII art previews: See your contribution skyline in the terminal while your STL file is being generated

![ascii-art-example](https://github.com/user-attachments/assets/849fcbc2-c692-4b10-9831-3de137d83bba)

✨ And, **it’s open source**! Give it a star, and contribute with feature suggestions or pull requests at [https://github.com/github/gh-skyline](https://github.com/github/gh-skyline).

Want to get started? Make sure you have the [GitHub CLI](https://cli.github.com/) installed!

Once you have that installed and are logged in, you can install the extension with `gh extension install github/gh-skyline`

Here are a few example commands to get you started:

Generate a skyline:  
`gh skyline --year 2024`

Generate a skyline for a specific user and year range:  
`gh skyline –-user chrisreddington --year 2014-2024`

Don’t have a 3D printer? No problem! Did you know that you can upload STL files to a GitHub repository and they’ll render directly in your browser?

![Skyline STL render in GitHub](https://github.com/user-attachments/assets/cd6152eb-72e3-46f1-b005-50ca3f86d6e5)

👀 We’d love to see your GitHub contribution history as a skyline! Drop a comment below with your render or 3D printed version, and let’s celebrate our contributions as a community!

**Comment by chrisreddington** [comment]
I'm excited to see everyone's contributions! I [uploaded an STL file of my contributions since I've joined GitHub](https://github.com/chrisreddington/chrisreddington/blob/master/chrisreddington-11-24-github-skyline.stl), and have definitely been a lot busier here over the last 4-5 years or so!

<img width="866" alt="image" src="https://github.com/user-attachments/assets/104ebdea-b8cf-4869-963f-bf618db2d531">

**Comment by martinwoodward** [comment]
I printed mine at 150% in a bronze filament on my Bambu X1C using 10% lightening infill - It show's my career at GitHub so far.

![martinwoodward contribution graph 2020-2024](https://github.com/user-attachments/assets/b9d38f66-300d-4246-a071-364aa010097c) 

While seeing some of the peaks brings back memories of all the cool things I've gotten to do in that time, I'm also proud that you can definitely see gaps for most weekends.  Also there is a big chunk where I took some time out this summer and it's very visible against my normally permanently online life.

**Comment by billythekid** [comment]
So happy to see the return of Skylines! I gifted a couple of 3D printed skylines from the old web version we had to friends as gifts and they loved them.

I particularly like how we can do a date range with this version, makes the cityscape vibe even better. 

<img width="830" alt="Screenshot 2024-12-09 at 17 20 29" src="https://github.com/user-attachments/assets/e83d0a3a-1440-46d9-8014-0c9021c5cf4e">

[My stl](https://github.com/billythekid/billythekid/blob/master/billythekid-18-24-github-skyline.stl)

Time to get my pals accounts loaded in to the ranges for some updated, and timely, gifts... 🤠

**Comment by colbyfayock** [comment]
Good stuff. Printed it bit scaled down on my Bambu A1

![IMG_0080 (1)](https://github.com/user-attachments/assets/2356e5c8-8cd4-4ba6-8e44-c9549e185c5a)

**Comment by RodrigoMvs123** [comment]
Hi ! I just fineshed the gh skyline -user... command. 
I already recieved a successfully message to: ser-21-24-github-skyline.stl. 
Please ! How can I share my Github Skyline on my Github user acccount or Linkedin Profile ?

**Reply by chrisreddington** [reply]
Thanks for your interest in `gh-skyline1 @RodrigoMvs123! 🎉 Glad to see you have already used the extension in the CLI!

You would need to open the `STL` file in a 3D Model Viewing app, take a screenshot and then upload it. Alternatively, you could upload the STL file to a GitHub repository (even a private one!). When you view the STL file in the browser, it will be rendered in 3D. You can [see an example in a repo of my own Skyline](https://github.com/chrisreddington/chrisreddington/blob/master/chrisreddington-11-24-github-skyline.stl).

Hope that helps! ❤️

**Reply by chrisreddington** [reply]
PS @RodrigoMvs123 - @aidansteele pointed out that there was a typo in the command in the initial post. It should be `--user` instead of `-user`. In the filename you outputted, it looks like you have generated it for `ser`, instead of your own use, so please try that again.

Many thanks to @aidansteele for [pointing that out](https://github.com/orgs/community/discussions/146467#discussioncomment-11529239)!

**Reply by RodrigoMvs123** [reply]
@chrisreddington Ok ! And where do I find my `STL` file in the github CLI extension ? 
Back on the `-user` I typed it wrong here in the chat but in the github CLI I did it right :)

**Reply by chrisreddington** [reply]
@RodrigoMvs123 It should be generated in the same directory that you executed the CLI command, hope that helps!

**Comment by foosel** [comment]
15 years of history 😬 And I'm more proud of the gaps than the peaks - they show I've learned to take breaks and recharge 💪

![image-12~2.png](https://github.com/user-attachments/assets/2ef51011-46cc-4942-ac60-0a310e9ed432)

**Comment by kamermans** [comment]
Nice, thanks!  Now I need to figure out which material to print it in :)

![image](https://github.com/user-attachments/assets/7604ec81-0ca3-4ca1-b7d4-4e1599027066)

**Comment by rfinnie** [comment]
Does this work with GitHub Enterprise Server? I've tried `gh auth switch` to a GHES account, and also tried `gh skyline` while in the directory of a GHES repo, and both times it just picked a github.com account.

**Reply by chrisreddington** [reply]
Hey @rfinnie, thanks for the interest in gh-skyline! It's a great question, would you mind raising an issue over [in the repository](https://github.com/github/gh-skyline) so that we can dig deeper into it? Thank you!

**Comment by aidansteele** [comment]
This looks great. Just FYI your example usage for a given user is probably misleading: it's missing a second hyphen in `--user`. The command as-is ends up generating an STL of activity for @ser. I can see (based on the filename) that at least one other user in this discussion has fallen victim to this typo.

**Reply by chrisreddington** [reply]
Excellent spot, thanks for pointing that out @aidansteele. I have corrected the post :)

**Comment by Chambras** [comment]
Could be a lot better that is for sure :) 
but not as flat as I thought it would be ;) 

![image](https://github.com/user-attachments/assets/7532221d-ca37-42ed-99fb-4c871bce4b28)

**Comment by kamermans** [comment]
@chrisreddington, I want to print these as gifts for my developers, but it is unable to get the contribution graph for private contributions.  This should be possible since they are in my organization, I'm an admin, and I can see their private contribution graph on their profile when I'm authenticated.

**Reply by kamermans** [reply]
Nevemind, I see this was already solved as an issue: https://github.com/github/gh-skyline/issues/16#issuecomment-2527692841

TL;DR: you need to re-authenticate with the `read:user` scope:
```
gh auth login --scopes read:user
```

**Reply by FrenkenFlores** [reply]
Hi! for public organizations you can use [gitprint.store](https://gitprint.store) and it's totally free. It can create a 3D model for a repo if you type-in for `<org_name>/<repo_name>`.
<img width="1093" height="961" alt="Screenshot from 2026-02-20 21-05-08" src="https://github.com/user-attachments/assets/71ce4ebb-22cc-46d0-9503-73b008436f15" />
